src/services/db/GitFileSystemService.ts

```diff
@@ -651,25 +649,32 @@ export default class GitFileSystemService {
             return okAsync(true)
           }
 
-          return ResultAsync.fromPromise(
-            this.git
-              .cwd({ path: `${efsVolPath}/${repoName}`, root: false })
-              .add(pathSpec),
-            (error) => {
-              logger.error(
-                `Error when Git adding files to ${repoName}: ${error}`
-              )
+          // Note: We need to add files sequentially due to the Git lock
```

Comment:
why can't we do a `git add .`?

Reply:
We don't have sufficient guarantees that doing a `git add .` only adds the files that were indeed changed by the user. There may be other changes that were made on the repo (which may be there for unknown reasons) but not yet committed, we don't want to accidentally add them here.